### PR TITLE
Added Dark mode & Light mode toggle button 

### PR DESCRIPTION
**Pull request by saravanasai**
# Added dark & light mode toggle buttons to show/hide the password field on the login page.

## Description

The PR is regarding adding a dark and light mode toggle button on the dropdown. Then added a show/hide the password field on the login page for a better user experience. I have used Alpin.js for reactivity.

## Related Issue
I made this PR to solve the #43 issue created by @saeedvaziry. This is my first contribution to this project I getting familiar with the code base. I hope I will do my best for this community in the future.

## Changes Made

1. I have added some small check box on the login page for implementing the show/hide password feature.
2. I have added a script on the `layout.app.blade` file to toggle the theme & added a button on the dropdown menu.
3. Added language for future internationalization 
4. Made changes on the tailwind config file to achieve theme changes

## Screenshots 

**Dark / Light mode feature** 

image

image

**Show / Hide password feature** 

image

image

### CODE REVIEW

File: lang/en/auth.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?php
```
Comment (saeedvaziry) [inline lang/en/auth.php:1]: I wouldn't use the old way translation files like this. at the moment no plan for supporting more languages. just trying to use `__("Translation")` method so later the adding another language will be easier with the language file json like `en.json`. So I would delete all files in the `lang` folder

File: resources/views/layouts/app.blade.php [hunk 2]
```diff
@@ -1,13 +1,14 @@
 @props(['server'])
 <!DOCTYPE html>
-<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">
+<html lang="{{ str_replace('_', '-', app()->getLocale()) }}" class="dark">
```
Comment (saeedvaziry) [inline resources/views/layouts/app.blade.php:3]: I wouldn't force the default schema to be dark. I would let it to be as the user's system by default unless they want to change it

File: resources/views/layouts/app.blade.php [hunk 3]
```diff
@@ -139,18 +167,58 @@
...
+                                                    clip-rule="evenodd" />
                                             </svg>
                                         </div>
                                     </x-slot>
 
                                     <x-slot name="content">
                                         <x-dropdown-link :href="route('profile')">Profile</x-dropdown-link>
                                         <div class="border-t border-gray-100 dark:border-gray-700"></div>
+                                        {{-- Dark Mode Toggle Button section  --}}
+                                        <x-dropdown-link x-data="{
```
Comment (saeedvaziry) [inline resources/views/layouts/app.blade.php:183]: personally I would rather to put the dark mode toggle next to the user's name on the top navigation bar to make it easier to access to it.

File: lang/en/auth.php (hunk 1 above)
Comment (saravanasai) [reply lang/en/auth.php:1]: Fine @saeedvaziry Let me change those accordingly & push the changes. Thanks for review

File: resources/views/layouts/app.blade.php (hunk 2 above)
Comment (saravanasai) [reply resources/views/layouts/app.blade.php:3]: @saeedvaziry  Hm. I got your point let me fix this too

File: resources/views/layouts/app.blade.php (hunk 3 above)
Comment (saravanasai) [reply resources/views/layouts/app.blade.php:183]: @saeedvaziry  Hm.  I didn't want to change the home page view. Let me place that toggle button on the top navigation bar near that username

**Comment by saravanasai** [comment]
@saeedvaziry  I have updated the PR as per your comments.Please review it once & share your comments

**Comment by saeedvaziry** [comment]
srcshot 2023-09-17 at 10 32 14

I guess you need to build assets to make it work

```
npm run build
```

And what do you think about this?

srcshot 2023-09-17 at 10 34 26@2x

And last one, I think would be better to have an option to also enable users to have their system preferences as the dark or light mode.

**Comment by saeedvaziry** [comment]
And one more, The page seems to be refreshing when changing the color scheme

**Comment by saravanasai** [comment]
> srcshot 2023-09-17 at 10 32 14 [ srcshot 2023-09-17 at 10 32 14 ](https://user-images.githubusercontent.com/61919774/268491353-255dcad8-ccdd-465f-a331-8189dc9aee30.gif) [ ](https://user-images.githubusercontent.com/61919774/268491353-255dcad8-ccdd-465f-a331-8189dc9aee30.gif)
> 
> I guess you need to build assets to make it work
> 
> ```
> npm run build
> ```
> 
> And what do you think about this?
> 
> srcshot 2023-09-17 at 10 34 26@2x
> 
> And last one, I think would be better to have an option to also enable users to have their system preferences as the dark or light mode.

Yes. I do not have any idea about the UI part to make users select system preferences.so I made it by default it use system preferences. If users want changes then they can use that toggle button to change the theme.

**Comment by saravanasai** [comment]
@saeedvaziry  Hi. I have fixed the build issues & refreshed on theme change & re-vamped the login page UI. On the system preference on the theme is the default it will be as system settings. In case the user is toggling the theme mode it works accordingly. Please share your comments on this PR & I am also planning to contribute to #26. I need some support for that I have seen the code. Triggering the bash script file on server creation. What all are versions can we support is there any limitation for it? I need a more clear idea of it to start work on it.

**Review by saeedvaziry — APPROVED** [review]

**Comment by saeedvaziry** [comment]
Merged. Thanks for the contribution.
Postgresql implementation is a bit complicated but not impossible. We need to follow the structure for the current ones (MySQL) and extend it for Postgresql as well.
Keep in mind that Backups also need to be covered by this.

**Pull request merged**